classes/core.php

```diff
@@ -823,6 +823,23 @@ public static function save_field_data( $field, $entry_id, $form ) {
 	}
 
 	/**
+	 * QCOSS-113: Clean magic tags value
+	 * Check if the processed magic value is same as the default value.
+	 * If yes Make it blank. Required when processor is linked to magic tags and due to some conditions
```

Comment:
@agilewarealok I 100% agree with your goal here. If the value was removed by conditional logic, return should be empty value.

But, what if the value should be default? IE it was never changed  from default.

Scenario I'm concerned about - field has default value of 10, I don't change that, so doesn't L834 evaluate true here?

Reply:
We've covered that case.
We will first check the field value by `explode_bracket_magic`, which checks if the value consists magic tags value or not. We will only compare new and old values if default value is magic tags value otherwise we return the processed value (Default value in this case)

Reply:
Ok, that makes sense. I will test with that in mind.

FYI I'm keeping this on 1.5.7 milestone. I'm hoping to get 1.5.6.2 out on Monday and keep it to calculation field improvements. I will merge this once 1.5.6.2 is tagged.

Reply:
Thanks @Shelob9 , Appreciate it.